internal/backends/winit/accesskit.rs

```diff
@@ -0,0 +1,471 @@
+
+        let win = self.window_adapter_weak.clone();
+        i_slint_core::timers::Timer::single_shot(Default::default(), move || {
+            if let Some(window_adapter) = win.upgrade() {
+                let self_ = &window_adapter.accesskit_adapter;
+                self_.inner.update_if_active(|| self_.build_new_tree())
+            };
+        });
+    }
+
+    fn add_node(&self, node: MappedNode) -> NodeId {
+        let index: usize = self.all_nodes.borrow().len();
+        let id = NodeId(
+            std::num::NonZeroU128::new(
+                (index as u128) << usize::BITS
+                    | (self.tree_generation.get() as u128 & usize::MAX as u128),
+            )
+            .unwrap(),
+        );
+        self.all_nodes.borrow_mut().push(node);
+        id
+    }
+
+    fn nodes_iter(&self) -> NodeIter<'_> {
+        NodeIter {
+            nodes: Some(std::cell::Ref::map(self.all_nodes.borrow(), |vec| &vec[..])),
+            index: 0,
+            tree_generation: self.tree_generation.get(),
+        }
+    }
+
+    fn item_rc_for_node_id(&self, id: NodeId) -> Option<ItemRc> {
+        let index: usize = (id.0.get() >> usize::BITS) as _;
+        let generation: usize = (id.0.get() & usize::MAX as u128) as _;
+        if generation != self.tree_generation.get() {
+            return None;
+        }
+        self.all_nodes.borrow().get(index).and_then(|cached_node| cached_node.item.upgrade())
+    }
+
+    fn find_node_id_by_item_rc(&self, mut item: ItemRc) -> Option<NodeId> {
+        while !item.is_accessible() {
+            if let Some(parent) = item.parent_item() {
+                item = parent;
+            } else {
+                break;
+            }
+        }
+
+        self.nodes_iter().find_map(|(id, cached_node)| {
+            cached_node.item.upgrade().and_then(|cached_item| (cached_item.eq(&item)).then(|| id))
+        })
+    }
+
+    fn rebuild_tree_of_dirty_nodes(&self) {
+        if !self.global_property_tracker.is_dirty() {
+            return;
+        }
+
+        // It's possible that we may have been triggered by a timer, but in the meantime
+        // the node tree has been emptied due to a tree structure change.
+        if self.all_nodes.borrow().is_empty() {
+            return;
+        }
+
+        let Some(window_adapter) = self.window_adapter_weak.upgrade() else { return };
+        let window = window_adapter.window();
+
+        self.inner.update_if_active(|| {
+            self.global_property_tracker.as_ref().evaluate_as_dependency_root(|| {
+                let scale_factor = ScaleFactor::new(window.scale_factor());
+
+                let nodes = self.nodes_iter().filter_map(|(id, cached_node)| {
+                    cached_node.tracker.as_ref().evaluate_if_dirty(|| {
+                        let item = cached_node.item.upgrade()?;
+
+                        let mut builder = self.build_node_without_children(&item, scale_factor);
```

Comment:
the scale_factor will not be taken as a dependency of the tracker because you query it before. Is that intentional?